Source/Core/Repositories/OrganizationRepository.cs

```diff
@@ -234,6 +226,21 @@ public class OrganizationRepository : RepositoryBase<Organization>, IOrganizatio
             return overLimit;
         }
 
+        private double GetTotalBlocked(Organization org, int count, double monthlyTotal, double monthlyBlocked, double hourlyTotal, double hourlyBlocked, bool applyHourlyLimit) {
+            if (org.IsSuspended)
+                return count;
+
+            int monthlyEventLimit = org.GetMaxEventsPerMonthWithBonus();
+            if ((monthlyTotal - monthlyBlocked) > monthlyEventLimit)
+                return monthlyTotal - monthlyBlocked - monthlyEventLimit;
+
+            int hourlyEventLimit = org.GetHourlyEventLimit();
+            if (applyHourlyLimit && (hourlyTotal - hourlyBlocked) > hourlyEventLimit)
+                return hourlyTotal - hourlyBlocked - hourlyEventLimit;
+
+            return count;
```

Comment:
I greatly simplified the logic from above. I think we were over thinking it.. The current logic passes all tests. Please give it a look and let me know.